### ISSUE TITLE
JSON formatting breaks with certain options enabled

### ISSUE THREAD

**Issue by felixjung** [sumup-oss/foundry#123]
## Describe the bug

When working in a TypeScript project with the ESLint `openSource` option turned on, the new JSON formatting causes problems, preventing the pre-commit hooks from passing whenever `package.json` is touched.

**Specifically**

* The copyright header linter is applied to `package.json`, which is wrong as you cannot add comments to JSON.
* The `parserOptions.project` configuration for the `@typescript-eslint/parser` starts to complain because a TypeScript project typically isn't configured to include JSON files.

Here is the log output.

```bash
husky > pre-commit (node v14.5.0)
✔ Preparing...
⚠ Running tasks...
  ❯ Running tasks for *.(js|jsx|json)
    ✖ foundry run eslint --fix [FAILED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up... 

✖ foundry run eslint --fix:
🚨 Executing the command "/Users/felix/Development/sumup-api-browser/node_modules/eslint/bin/eslint.js --fix /Users/felix/Development/sumup-api-browser/package.json /Users/felix/Development/sumup-api-browser/tsconfig.json" failed
🚨 Error: /Users/felix/Development/sumup-api-browser/node_modules/eslint/bin/eslint.js exited with an error (code 1).

/Users/felix/Development/sumup-api-browser/package.json
  1:2  error  Missing notice header  notice/notice

/Users/felix/Development/sumup-api-browser/tsconfig.json
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: tsconfig.json.
The extension for the file (.json) is non-standard. You should add "parserOptions.extraFileExtensions" to your config

✖ 2 problems (2 errors, 0 warnings)

🚨 Executing the command "/Users/felix/Development/sumup-api-browser/node_modules/lint-staged/bin/lint-staged.js " failed
🚨 Error: /Users/felix/Development/sumup-api-browser/node_modules/lint-staged/bin/lint-staged.js exited with an error (code 1).
husky > pre-commit hook failed (add --no-verify to bypass)
```

## Steps to reproduce

1. Configure a typescript project with Foundry and label it as OpenSource.
2. Make changes to the `package.json`
3. Attempt to commit the changes.

## Expected behavior

Committing JSON files should work out of the box in a TypeScript open source project. The copyright header formatting should not be applied to JSON files and probably ESLint should not be applied either. Ideally, only prettier would run on these files.

## Specifications

- Package version: `^3.1.0`
- Node version: `14.5.0`
- OS name & version: macOS Catalina

## Additional context

Here are the various configuration files in my project.

* `.eslintrc.js`

```javascript
module.exports = require('@sumup/foundry/eslint')({
  language: 'TypeScript',
  environments: ['Browser'],
  frameworks: ['React', 'Emotion', 'Jest'],
  openSource: true,
});
```

* `prettier.config.js`

```javascript
module.exports = require('@sumup/foundry/prettier')({
  proseWrap: 'always',
  overrides: [
    {
      files: '*.md',
      options: { parser: 'markdown' }
    },
    {
      files: '*.yaml',
      options: { parser: 'yaml' }
    }
  ]
});
```

* `tsconfig.json`

```json
{
  "compilerOptions": {
    "target": "ES2020",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "baseUrl": "./src",
    "allowJs": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "declaration": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "sourceMap": true,
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    "**/*.js",
  ],
  "exclude": [
    "node_modules"
  ]
}
```

**Comment by felixjung** [comment]
Trying the canary release now.

**Comment by felixjung** [comment]
Unfortunately not fixed on canary.

**Comment by connor-baer** [comment]
@felixjung Is this fully resolved by #124? You said you needed to make further changes to your `tsconfig`, right?
